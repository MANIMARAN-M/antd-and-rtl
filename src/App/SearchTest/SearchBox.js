import React from "react";
import { useState } from "react";

const SearchBox = () => {
  const [query, setQuery] = useState("");
  const searhClick = () => {
    if (query) {
      console.log(query);
    }
  };
  const updateQuery = (e) => {
    setQuery(e.target.value);
  };
  return (
    <div>
      <input
        type="text"
        name="search"
        placeholder="Search"
        onChange={updateQuery}
        value={query}
      />
      <button onClick={searhClick} data-testid="search-button">
        Search
      </button>
    </div>
  );
};

export default SearchBox;
