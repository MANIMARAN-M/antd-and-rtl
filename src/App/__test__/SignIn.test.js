import { render, fireEvent, cleanup } from "@testing-library/react";
import SearchBox from "../SearchTest/SearchBox";

afterEach(cleanup);
it("rendered correctly", () => {
  const { queryAllByTestId, queryByPlaceholderText } = render(<SearchBox />);
  expect(queryAllByTestId("search-button")).toBeTruthy();
  expect(queryByPlaceholderText("Search")).toBeTruthy();
});
test("Sample test", () => {
  expect(2 + 2).toBe(4);
});
describe("Input Values", () => {
  it("update on change", () => {
    const { queryByPlaceholderText } = render(<SearchBox />);
    const SearchInput = queryByPlaceholderText("Search");
    fireEvent.change(SearchInput, { target: { value: "test" } });
    expect(SearchInput.value).toBe("test");
  });
});
describe("Search button", () => {
  it("does not trigger requestSearch function", () => {
    const requestSearch = jest.fn();
    const { queryByTestId } = render(
      <SearchBox requestSearch={requestSearch} />
    );
    fireEvent.click(queryByTestId("search-button"));
    expect(requestSearch).not.toHaveBeenCalled();
  });
});
describe("with data inside the query", () => {
  it("with data inside the querys", () => {
    const requestSearch = jest.fn();
    const { queryByTestId, queryByPlaceholderText } = render(
      <SearchBox requestSearch={requestSearch} />
    );
    const SearchInput = queryByPlaceholderText("Search");
    fireEvent.change(SearchInput, { target: { value: "test" } });
    fireEvent.click(queryByTestId("search-button"));
    expect(SearchInput.value).toBe("test");
    // expect(requestSearch).toHaveBeenCalledTimes(1);
  });
});
describe("take a snapshot", () => {
  it("should take a snapshot", () => {
    const { asFragment } = render(<SearchBox />);
    expect(asFragment(<SearchBox />)).toMatchSnapshot();
  });
});
