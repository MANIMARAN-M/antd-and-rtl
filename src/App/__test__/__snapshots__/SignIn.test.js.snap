// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`take a snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <input
      name="search"
      placeholder="Search"
      type="text"
      value=""
    />
    <button
      data-testid="search-button"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
